crates/nu-command/src/system/run_external.rs

```diff
@@ -321,37 +325,75 @@ fn expand_tilde(arg: &str) -> String {
 ///
 /// Note: This matches the default behavior of Bash, but is known to be
 /// error-prone. We might want to change this behavior in the future.
-fn expand_glob(arg: &str, cwd: &Path, span: Span) -> Result<Vec<String>, ShellError> {
-    let Ok(paths) = nu_glob::glob_with_parent(arg, nu_glob::MatchOptions::default(), cwd) else {
+fn expand_glob(
+    arg: &str,
+    cwd: &Path,
+    span: Span,
+    interrupt: &Option<Arc<AtomicBool>>,
+) -> Result<Vec<String>, ShellError> {
+    const GLOB_CHARS: &[char] = &['*', '?', '['];
+
+    // Don't expand something that doesn't include the GLOB_CHARS
+    if !arg.contains(GLOB_CHARS) {
+        return Ok(vec![arg.into()]);
+    }
+
+    // We must use `nu_engine::glob_from` here, in order to ensure we get paths from the correct
+    // dir
+    let glob = NuGlob::Expand(arg.to_owned()).into_spanned(span);
+    let Ok((_prefix, paths)) = nu_engine::glob_from(&glob, cwd, span, None) else {
+        // If an error occurred, return the original input
         return Ok(vec![arg.into()]);
     };
 
-    let mut result = vec![];
-    for path in paths {
-        let path = path.map_err(|err| ShellError::IOErrorSpanned {
-            msg: format!("{}: {:?}", err.path().display(), err.error()),
-            span,
-        })?;
-        // Strip PWD from the resulting paths if possible.
-        let path_stripped = if let Ok(remainder) = path.strip_prefix(cwd) {
-            // If stripping PWD results in an empty path, return `.` instead.
-            if remainder.components().next().is_none() {
-                Path::new(".")
+    // If the first component of the original `arg` string path was '.', that should be preserved
+    let relative_to_dot = Path::new(arg).starts_with(".");
```

Comment:
I don't think you need to handle these relative logic by yourself.  I think the logic before rewrite is good to use.

https://github.com/nushell/nushell/blob/c61075e20e328bc33f4765feb6d5fcef5e6039ce/crates/nu-command/src/system/run_external.rs#L623-L640

Reply:
Ok I can test whether it passes my tests

Reply:
I won't be able to check this until tomorrow, but I'm still interested in trying it